src/distributed_graph_generator.cc

```diff
@@ -402,6 +421,29 @@ void distributed_graph_generator::generate_distributed_commands(const task& tsk)
 		}
 	}
 
+	// Check for and report overlapping writes between local chunks, and between local and remote chunks.
+	if(m_overlapping_write_policy != error_policy::ignore) {
+		// Since this check is run distributed on every node, we avoid quadratic behavior by only checking for conflicts between all local chunks and the
+		// region-union of remote chunks. This way, every conflict will be reported by at least one node.
+		const box<3> global_chunk(subrange(full_chunk.offset, full_chunk.range));
+		auto remote_chunks = region_difference(global_chunk, region(box_vector<3>(local_chunks))).into_boxes();
+
+		// detect_overlapping_writes takes a single box_vector, so we concatenate local and global chunks (the order does not matter)
+		auto distributed_chunks = std::move(remote_chunks);
+		distributed_chunks.insert(distributed_chunks.end(), local_chunks.begin(), local_chunks.end());
+
+		if(const auto overlapping_writes = detect_overlapping_writes(tsk, distributed_chunks); !overlapping_writes.empty()) {
+			auto error = fmt::format("Task T{}", tsk.get_id());
+			if(!tsk.get_debug_name().empty()) { fmt::format_to(std::back_inserter(error), " \"{}\"", tsk.get_debug_name()); }
+			error += " has overlapping writes between multiple nodes in";
```

Comment:
If I'm not mistaken, there should never be more than one CDAG chunk per node in stable Celerity. Oversubscription will be handled entirely on the IDAG level (even though the graph generator uses lingo that suggests any number of chunks per node). I worded it like this explicitly because the equivalent message the `instruction_graph_generator` throws will talk about overlaps between local chunks / devices.

Reply:
Ah, right. So `box_vector<3> local_chunks;` really is a `box<3> local_chunk;`. We should adjust the language at some point.

Reply:
Thinking about this some more, there might actually be situations where nodes will be assigned multiple chunks in the future (for load balancing). But for now the message is correct!

Reply:
Mhh, I did not consider load balancing! What should the wording be if we imagine the IDAG level here as well? "multiple cluster chunks" is hard to understand as a user who does not know the internals of our work assignment.